subprojects/file-watching/src/main/java/org/gradle/internal/watch/registry/impl/AbstractFileWatcherUpdater.java

```diff
@@ -165,4 +180,20 @@ private static boolean isMissing(FileSystemLocationSnapshot snapshot) {
         // Missing accessed indirectly means we have a dangling symlink in the directory, and that's content we cannot ignore
         return snapshot.getType() == FileType.Missing && snapshot.getAccessType() == FileMetadata.AccessType.DIRECT;
     }
+
+    public interface MovedWatchedDirectoriesSupplier {
```

Comment:
This name is weird. Yes, this thing "supplies" moved directories, but that's just part of what it does. I think we should call this a `MovedDirectoryHandler` or something similar.

The method name should be in-line with the type name, too, i.e. if we call this `<Something>Directory<Something>` then we should call the method `stopWatchingMovedDirectories()` not `*Paths()`.

I'm also fine with going with `MovedPathHandler` and keeping the current directory name.

Reply:
`MovedDirectoryHandler` it is.